yjit/src/core.rs

```diff
@@ -183,6 +183,16 @@ impl Type {
         }
     }
 
+    /// Returns an Option boolean representing whether the value is equal to nil if known, otherwise None
+    pub fn known_nil_p(&self) -> Option<bool> {
```

Comment:
Small detail, but I think we should name this `known_nil` because usually `nil_p` is for a predicate function which this doesn't behave like other predicate functions with `_p` in the codebase. It would also be more in line with the other `known_x` methods.

Reply:
👍 Sounds good